libraries/provider_rightscale_volume.rb

```diff
@@ -556,7 +556,15 @@ def attach_volume(volume_id, device)
         scan_for_attachments if node['virtualization']['system'] == 'vmware'
 
         # Determine the actual device name
-        (Set.new(get_current_devices) - current_devices).first
+        added_device = nil
+        Timeout::timeout(60) do
+          begin
+            sleep(1)
+            Chef::Log.info "Checking for added device."
+            added_device = (Set.new(get_current_devices) - current_devices).first
+          end until !added_device.nil?
+        end
```

Comment:
In that case you would need another line before:
```ruby
added_device = nil
```
I think a better simplification for the condition would be to do:
```ruby
begin
  ...
end until added_device
```